### PR TITLE
add kotlin IM read test and enable im test in CI

### PR DESCRIPTION
**Pull request by yunhanw-google**
--add kotlin IM read test and enable im test in CI
--use kotlin tlvwriter in kotlin im write/invoke test.

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -601,6 +601,54 @@ jobs:
                      --tool-args "onnetwork-long --nodeid 1 --setup-pin-code 20202021 --discriminator 3840 -t 1000" \
                      --factoryreset \
                   '
+              - name: Run IM Invoke Test
+              timeout-minutes: 10
```
Comment (yufengwangca) [inline .github/workflows/tests.yaml:605]: Could we change the timeout to 1, 10 mins is too long for a IM command.

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -601,6 +601,54 @@ jobs:
...
+                  './scripts/tests/run_java_test.py \
+                     --app out/linux-x64-all-clusters-ipv6only-no-ble-no-wifi-tsan-clang-test/chip-all-clusters-app \
+                     --app-args "--discriminator 3840 --interface-id -1" \
+                     --tool-path out/linux-x64-java-matter-controller \
+                     --tool-cluster "pairing" \
+                     --tool-args "onnetwork-long-im-invoke --nodeid 1 --setup-pin-code 20202021 --discriminator 3840 -t 1000" \
+                     --factoryreset \
+                  '
+              - name: Run IM Read Test
+              timeout-minutes: 10
```
Comment (yufengwangca) [inline .github/workflows/tests.yaml:617]: Same, could we change the timeout to 1

File: .github/workflows/tests.yaml [hunk 3]
```diff
@@ -601,6 +601,54 @@ jobs:
...
+                  './scripts/tests/run_java_test.py \
+                     --app out/linux-x64-all-clusters-ipv6only-no-ble-no-wifi-tsan-clang-test/chip-all-clusters-app \
+                     --app-args "--discriminator 3840 --interface-id -1" \
+                     --tool-path out/linux-x64-java-matter-controller \
+                     --tool-cluster "pairing" \
+                     --tool-args "onnetwork-long-im-read --nodeid 1 --setup-pin-code 20202021 --discriminator 3840 -t 1000" \
+                     --factoryreset \
+                  '
+              - name: Run IM Write Test
+              timeout-minutes: 10
```
Comment (yufengwangca) [inline .github/workflows/tests.yaml:629]: Same above

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -601,6 +601,54 @@ jobs:
...
+                  './scripts/tests/run_java_test.py \
+                     --app out/linux-x64-all-clusters-ipv6only-no-ble-no-wifi-tsan-clang-test/chip-all-clusters-app \
+                     --app-args "--discriminator 3840 --interface-id -1" \
+                     --tool-path out/linux-x64-java-matter-controller \
+                     --tool-cluster "pairing" \
+                     --tool-args "onnetwork-long-im-write --nodeid 1 --setup-pin-code 20202021 --discriminator 3840 -t 1000" \
+                     --factoryreset \
+                  '
+              - name: Run IM Subscribe Test
+              timeout-minutes: 10
```
Comment (yufengwangca) [inline .github/workflows/tests.yaml:641]: Same above

**Review by yufengwangca — APPROVED** [review]

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (yunhanw-google) [reply .github/workflows/tests.yaml:605]: offline chat, it has pairing test inside, so we align the timeout 10min with the previous pairing test.

File: .github/workflows/tests.yaml (hunk 3 above)
Comment (yunhanw-google) [reply .github/workflows/tests.yaml:629]: offline chat, it has pairing test inside, so we align the timeout 10min with the previous pairing test.

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (yunhanw-google) [reply .github/workflows/tests.yaml:617]: offline chat, it has pairing test inside, so we align the timeout 10min with the previous pairing test.

File: .github/workflows/tests.yaml (hunk 4 above)
Comment (yunhanw-google) [reply .github/workflows/tests.yaml:641]: offline chat, it has pairing test inside, so we align the timeout 10min with the previous pairing test.

**Pull request merged**